exercises/concept/gotta-snatch-em-all/.meta/design.md

```diff
@@ -0,0 +1,74 @@
+# Design
+
+## Learning objectives
+
+- Get to know the `Set` interface.
+- Know about the `HashSet` class.
+- Know how the `add` method works.
+- Know how to perform set operations like asymmetric difference, intersection and union.
+
+## Out of scope
+
+- More specific subtypes of `Set`, such as `OrderedSet` and `SequencedSet`.
+- More advanced classes implementing the `Set` interface, such as `TreeSet` and `LinkedHashSet`.
+
+## Concepts
+
+- `sets`
+
+## Prerequisites
+
+This exercise's prerequisites Concepts are:
+
+- `lists`
+- `generic-types`
+
+## Analyzer
+
+This exercise could benefit from the following rules in the [analyzer].
+If the solution does not receive any of the listed feedback, it must be exemplar.
+Leave a `celebratory` comment to celebrate the success!
+
+### Task 1
+
+The idea behind this task is that the student knows how to convert from a `List` to a `Set`.
+Ideally we want the student to use the `HashSet(Collection)` constructor.
+If the solution instead creates a new empty set and manually `add` or `addAll` the items in the list, leave an `actionable` comment pointing to the `HashSet` constructor instead.
+
```

Comment:
Well this was a bit of an experiment, but it was also easy to provide much more detail because I was busy creating the exercise.

IMO the more information we can provide the better, because this basically nails down the requirements for the analyzer implementation. Any ambiguity may therefore lead to bugs in the analyzer or a long-winded review process when the analyzer is implemented.

I don't necessarily want to update all of the existing design documents to change to a different format. First of all, I think it's fine if these documents diverge in formatting a bit, as long as they provide the necessary context and information for their respective exercises. Second, it'll probably be harder to go into this much detail for exercises we didn't author ourselves. But if you have specific exercises in mind where you think a format like this might make the analyzer section more clear, by all means open a PR to address it! 😄

Reply:
👍 Perfect